test/e2e/validation_test.go

```diff
@@ -667,22 +665,24 @@ func (tc *testContext) findNodeCSRs(nodeName string) ([]certificates.Certificate
 
 // validateUpgradeableCondition ensures that the operator's Upgradeable condition is correctly communicated to OLM
 func (tc *testContext) validateUpgradeableCondition(expected meta.ConditionStatus) error {
-	ocName, present := os.LookupEnv(condition.OperatorConditionName)
-	if !present {
-		// Implies operator is not OLM-managed
-		return nil
-	}
-	err := wait.Poll(retry.Interval, retry.ResourceChangeTimeout, func() (bool, error) {
-		oc := &operators.OperatorCondition{}
-		err := tc.client.Cache.Get(context.TODO(), types.NamespacedName{Namespace: tc.namespace, Name: ocName}, oc)
+	deployment, err := tc.client.K8s.AppsV1().Deployments(resourceNamespace).Get(context.TODO(), resourceName,
+		meta.GetOptions{})
+	if err != nil {
+		log.Printf("error getting operator deployment: %v", err)
+		return err
+	}
+	// Get the operator condition name using the deployment spec
+	ocName := deployment.Spec.Template.Spec.Containers[0].Env[3].Value
+	if ocName == "" {
+		return errors.Errorf("%s must be set for OLM managed operators", condition.OperatorConditionName)
+	}
+	err = wait.Poll(retry.Interval, retry.ResourceChangeTimeout, func() (bool, error) {
+		oc, err := tc.client.Olm.OperatorsV1().OperatorConditions(tc.namespace).Get(context.TODO(), ocName, meta.GetOptions{})
 		if err != nil {
 			log.Printf("unable to get OperatorCondition %s from namespace %s", ocName, tc.namespace)
 			return false, nil
 		}
-
-		specCheck := condition.Validate(oc.Spec.Conditions, operators.Upgradeable, expected)
-		statusCheck := condition.Validate(oc.Status.Conditions, operators.Upgradeable, expected)
-		return specCheck && statusCheck, nil
+		return condition.Validate(oc.Status.Conditions, operators.Upgradeable, expected), nil
```

Comment:
Any reason to move to away from checking both spec and status? I felt that checking both was important since the upgradeable condition changes so often, we should ensure it has settled at the expected state (i.e. `spec == status == expected`).

Reply:
good point, I was thinking why it wasn't available in the object. I realised I was using the v1 client, switched to v2 now and I see conditions in spec now. Updating this as well.